### ISSUE TITLE
Allow rebalance question with manual commits

### ISSUE THREAD

**Issue by madshov** [twmb/franz-go#634]
Hey twmb,

I'm looking at the goroutine-per-partition-consuming example with manual commit. Could there be any problems in calling AllowRebalance() before all fetched records have been committed? I think that could happen in the example https://github.com/twmb/franz-go/blob/master/examples/goroutine_per_partition_consuming/manual_commit/main.go. What would happen if a broker decided to rebalance while some records are still being processed and havent been committed yet? As far as I can see, AllowRebalance() is called once every record has been sent to each consuming goroutine (ie. not committed yet).
```
func (s *splitConsume) poll(cl *kgo.Client) {
	for {
		fetches := cl.PollRecords(context.Background(), 10000)
		if fetches.IsClientClosed() {
			return
		}
		fetches.EachPartition(func(p kgo.FetchTopicPartition) {
			tp := tp{p.Topic, p.Partition}
			s.consumers[tp].recs <- p.Records
		})
		cl.AllowRebalance()
	}
}
```
In other words, what happens if lines 51 and 52 takes several minutes for each record, and a rebalance is triggered in the mean time?
```
time.Sleep(time.Duration(rand.Intn(150)+100) * time.Millisecond) // simulate work
fmt.Printf("Some sort of work done, about to commit t %s p %d\n", pc.topic, pc.partition)
```
Hope it makes sense.

**Comment by twmb** [comment]
That does make sense. But, this is guarded because of line 94. A rebalance _can_ happen, but then the client will enter OnRevoked. If a partition is being lost by the client, line 91 kills the consumer, and line 94 defers a wait for the consumer to exit. So, _if_ records are being processed (lines 51 and 52), the consumer will _not exit_ until the records are committed (line 53).

However, this comes with the same downsides as BlockRebalanceOnPoll: if processing or committing takes too long, the consumer will be booted from the group. This is outright unavoidable as a group consumer -- so, although it is niche, it is impossible to prevent _every_ possibility of duplicate processing, per protocol. But, you can try to engineer around it at your risk tolerance level (and this example does show a relatively paranoid level of engineering, although it is a small amount of code).
